src/chunk_insert_state.c

```diff
@@ -149,6 +150,19 @@ chunk_insert_state_create(Chunk *chunk, ChunkDispatch *dispatch)
 	if (parse != NULL && parse->onConflict != NULL)
 		state->arbiter_indexes = chunk_infer_arbiter_indexes(rti, dispatch->estate->es_range_table, dispatch->parse);
 
+	/* Set tuple conversion map, if tuple needs conversion */
+	parent_rel = heap_open(chunk_get_parent_oid(chunk->table_id), AccessShareLock);
```

Comment:
Shouldn't the parent here be dispatch->hypertable->main_table_relid. I think we can avoid a lookup here, no?